src/client/visualizations/line-chart/line-chart.tsx

```diff
@@ -553,7 +554,23 @@ export class LineChart extends BaseVisualization<LineChartState> {
         newState.continuousDimension = continuousDimension;
 
         var axisRange = essence.getEffectiveFilter(LineChart.id).getExtent(continuousDimension.expression) as PlywoodRange;
-        axisRange = axisRange ? axisRange : this.getXAxisRange(essence, continuousDimension, dataset);
+        if (axisRange) {
+          // Special treatment for realtime data, i.e. time data where the maxTime is within Duration of the filter end
```

Comment:
Is it possible to put this into a separate function ? The function's name would act as a comment :)